fittable.R

```diff
@@ -0,0 +1,36 @@
+fittable <- function(..., indices="robust", all=F) {
+  require(lavaan)
+  models <- list(...)
+  indic.names <- c("x2", "df", "p", "BIC", "CFI", "TLI", "RMSEA", "lower.RMSEA", "upper.RMSEA", "SRMR")
+  
+  if (indices == "scaled") {
+    indic <- c("chisq.scaled", "df.scaled", "pvalue.scaled", "bic", "cfi.scaled", "tli.scaled", "rmsea.scaled",
+               "rmsea.ci.lower.scaled", "rmsea.ci.upper.scaled", "srmr")
+
+  } else if (indices == "robust") {
+    indic <- c("chisq.scaled", "df.scaled", "pvalue.scaled", "bic", "cfi.robust", "tli.robust", "rmsea.robust",
+               "rmsea.ci.lower.robust", "rmsea.ci.upper.robust", "srmr")
+  } else {
+    indic.names <- indic <- indices
+  }
+
+  table.fit <- NULL
+  all.fit <- list()
+  for (m in c(1:length(models))) {
+    all.fit[[m]] <- fitMeasures(models[[m]])
+    fits2 <- all.fit[[m]][indic]
+    table.fit <- rbind(table.fit, fits2)
+  }
+  rownames(table.fit) <- paste("model", c(1:length(models)))
+  colnames(table.fit) <- indic.names
+  names(all.fit) <- paste0("model", c(1:length(models)))

```

Comment:
Hele, co takhle na to jit trochu jinak:

- Vyexportuju si vsechny indices ze vsech modelu do `df` misto do listu (all.fit)
- V druhem kroku vytvorim druhy `df` ktery bude filtrovat to prvni `df` (table.fit):

```
all.fit <- NULL

for (model in 1:length(models)){

  all.fit <- rbind(all.fit, fitMeasures(models[[model]]))

}

rownames(all.fit) <- paste('model', c(1:length(models)))
table.fit <- all.fit[, indic]
colnames(table.fit) <- indic.names
```

Reply:
Ha! Dobře ty!